File: src/include/asset/ScrapedAssetStatus.php
<?php

namespace asset;

enum ScrapedAssetStatus: int
{
	case NEWLY_FOUND = 0;
	case UPDATED = 1;
	case VALIDATED = 2;

	case NEWLY_FOUND_FAILED = 100;
	case UPDATED_FAILED = 101;
	case VALIDATED_FAILED = 102;

	public function toStoredAssetStatus(): StoredAssetStatus
	{
		return match ($this) {
			ScrapedAssetStatus::NEWLY_FOUND,
			ScrapedAssetStatus::UPDATED,
			ScrapedAssetStatus::VALIDATED => StoredAssetStatus::ACTIVE,
			ScrapedAssetStatus::NEWLY_FOUND_FAILED,
			ScrapedAssetStatus::UPDATED_FAILED => StoredAssetStatus::VALIDATION_FAILED_RECENTLY,
			ScrapedAssetStatus::VALIDATED_FAILED => StoredAssetStatus::VALIDATION_FAILED_RECENTLY,
		};
	}
}

File: app/asset/scraped_asset_status.py
from enum import IntEnum
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from app.asset.stored_asset_status import StoredAssetStatus


class ScrapedAssetStatus(IntEnum):
    NEWLY_FOUND = 0
    UPDATED = 1
    VALIDATED = 2
    NEWLY_FOUND_FAILED = 100
    UPDATED_FAILED = 101
    VALIDATED_FAILED = 102

    def to_stored_asset_status(self) -> "StoredAssetStatus":
        from app.asset.stored_asset_status import StoredAssetStatus

        success_statuses = {
            ScrapedAssetStatus.NEWLY_FOUND,
            ScrapedAssetStatus.UPDATED,
            ScrapedAssetStatus.VALIDATED,
        }
        if self in success_statuses:
            return StoredAssetStatus.ACTIVE
        return StoredAssetStatus.VALIDATION_FAILED_RECENTLY
